internal/shader/shader.go

```diff
@@ -500,24 +497,24 @@ func (s *compileState) parseMultipleVariable(stmts *[]shaderir.Stmt, i int, t *s
 		if t.Main == shaderir.None {
 			ts, ok := s.functionReturnTypes(block, init)
 			if ok {
-				inittypes = ts
+				*inittypes = ts
 			}
 			if len(ts) != len(vs.Names) {
 				s.addError(vs.Pos(), fmt.Sprintf("the numbers of lhs and rhs don't match"))
-				return initexprs, true
+				return *initexprs, true
 			}
 		}
 	}
 
-	if t.Main == shaderir.None && len(inittypes) > 0 {
-		*t = inittypes[i]
+	if t.Main == shaderir.None && len(*inittypes) > 0 {
+		*t = (*inittypes)[i]
 	}
 
-	if len(inittypes) > 0 && len(initexprs) > 0 && !canAssign(t, &inittypes[i], initexprs[i].Const) {
-		s.addError(vs.Pos(), fmt.Sprintf("cannot use type %s as type %s in variable declaration", inittypes[i].String(), t.String()))
+	if !canAssign(t, &(*inittypes)[i], (*initexprs)[i].Const) {
```

Comment:
I think this changes the behavior. Are you sure inittypes and initexprs cannot be empty here?

Reply:
The len check was a band-aid fix by me to solve another thing I hadn't implemented correctly. Your comment about the variables being outside the for-loop made me realize my mistake and now inttypes and intexprs can't be empty anymore. I haven't taken the time to prove it formally, but `case len(vs.Values) == 0:`@ 548 should guarantee it. 

Tests run and that's what matter (;